subsys/bluetooth/host/keys.c

```diff
@@ -371,3 +410,27 @@ SETTINGS_STATIC_HANDLER_DEFINE(bt_keys, "bt/keys", NULL, keys_set, keys_commit,
 			       NULL);
 
 #endif /* CONFIG_BT_SETTINGS */
+
+#if defined(CONFIG_BT_KEYS_OVERWRITE_OLDEST)
+void bt_keys_notify_usage(u8_t id, const bt_addr_le_t *addr)
+{
+	if (!addr) {
+		return;
+	}
+	struct bt_keys *keys = bt_keys_find_addr(id, addr);
+	if (!keys) {
+		return;
+	}
+	if (last_keys_updated == keys) {
+		return;
+	}
+	keys->aging_counter = ++aging_counter_val;
+	last_keys_updated = keys;
+	BT_ERR("Aging counter for %s is set to %u", bt_addr_le_str(addr),
+	       keys->aging_counter);
+#if (defined(CONFIG_BT_KEYS_SAVE_AGING_COUNTER_ON_PAIRING) && \
+	defined(CONFIG_BT_SETTINGS))
```

Comment:
The test for `CONFIG_BT_SETTINGS` seems redundant here since the Kconfig option already `depends on BT_SETTINGS`